internal/daemon/worker/session/session.go

```diff
@@ -288,11 +295,15 @@ func (s *sess) RequestCancel(ctx context.Context) error {
 }
 
 func (s *sess) RequestActivate(ctx context.Context, tofu string) error {
+	if s.GetStatus() == pbs.SESSIONSTATUS_SESSIONSTATUS_ACTIVE {
```

Comment:
This is made worse by the fact that `ApplyLocalStatus` and `ApplyLocalTofuToken` are locked calls so you could mark the status as active and have something read that before the tofu token is stored. Not sure if that ends up mattering. But I do think that there might be deeper issues with this pattern of constantly calling locked functions instead of just grabbing the lock once and using it.

Reply:
I guess the downside to an up-front lock is not wanting to hold that lock during the RPC call.

Reply:
I'll remove this since activateSession will simply return an already activated session